### PR TITLE
Benchmarks and tests code cleanups.

### PR DESCRIPTION
**Pull request by jerome-benoit**
Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] If your changes contain any new feature please be sure to document it.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

### CODE REVIEW

**Review by Shinigami92 — APPROVED** [review]

**Comment by Shinigami92** [comment]
> Thanks, there's also the README.MD ;)

Not on my machine :thinking:

File: benchmarks/internal/choose-worker.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 const suite = new Benchmark.Suite()
 
-const LIST_FORMATTER = new Intl.ListFormat('en-US', {
-  style: 'long',
-  type: 'conjunction'
-})
+function generateWorkersArray (numberOfWorkers) {
+  const workers = []
+  for (let i = 0; i < numberOfWorkers; i++) {
+    workers.push(i)
```
Comment (Shinigami92) [inline benchmarks/internal/choose-worker.js:9]: **nitpick (non-blocking):** Isn't this the same as `[...Array(numberOfWorkers).keys()]`?

Comment (jerome-benoit) [reply benchmarks/internal/choose-worker.js:9]: If that works with plain JS, you can push that change once tested ;)

Comment (Shinigami92) [reply benchmarks/internal/choose-worker.js:9]: Okay, currently running one benchmark-run.
Seems that it takes a while :watch:

**Comment by Shinigami92** [comment]
One Benchmark run on my machine:
image

Operating System: Arch Linux
KDE Plasma Version: 5.21.2
KDE Frameworks Version: 5.79.0
Qt Version: 5.15.2
Kernel Version: 5.10.20-1-lts
OS Type: 64-bit
Graphics Platform: X11
Processors: 28 × Intel® Core™ i9-10940X CPU @ 3.30GHz
Memory: 62.5 GiB of RAM
Graphics Processor: GeForce GTX 1080/PCIe/SSE2

**Comment by jerome-benoit** [comment]
> One Benchmark run on my machine:
> image
> 
> Operating System: Arch Linux
> KDE Plasma Version: 5.21.2
> KDE Frameworks Version: 5.79.0
> Qt Version: 5.15.2
> Kernel Version: 5.10.20-1-lts
> OS Type: 64-bit
> Graphics Platform: X11
> Processors: 28 × Intel® Core™ i9-10940X CPU @ 3.30GHz
> Memory: 62.5 GiB of RAM
> Graphics Processor: GeForce GTX 1080/PCIe/SSE2

I think there's a little confusion: the changes made here impact the internal benchmarking code: `npm run benchmark` to test ;)

**Comment by Shinigami92** [comment]
> I think there's a little confusion: the changes made here impact the internal benchmarking code: `npm run benchmark` to test ;)

Ah good to know, just saw some files that are also changed within `versus-external-pools` but didn't checked what really changes in terms of behavior

**Comment by Shinigami92** [comment]
image

**Comment by jerome-benoit** [comment]
> > Thanks, there's also the README.MD ;)
> 
> Not on my machine 🤔
> 

Event here: benchmarks/versus-external-pools/README.MD?

**Comment by Shinigami92** [comment]
image

**Comment by Shinigami92** [comment]
If @pioardi approves, you can merge it :slightly_smiling_face:

**Review by pioardi — APPROVED** [review]
Good job thanks and Sorry if on lady days I am not so Active

**Pull request merged**